settings.yaml

```diff
@@ -56,4 +56,7 @@ sagemaker:
 
 openai:
   api_key: ${OPENAI_API_KEY:}
-  model: gpt-3.5-turbo
\ No newline at end of file
+  model: gpt-3.5-turbo
+
+rag:
+  default_context_template: "Context information is below.\n--------------------\n{context_str}\n--------------------\n"
```

Comment:
Please use `|` (c.f. https://github.com/imartinez/privateGPT/pull/1403)

Reply:
Thanks for the comment. 

Happy to do so, just one thing. In this case I was trying to keep the value as closely as possible as the original one, that I understand includes "\n" (see https://github.com/run-llama/llama_index/blob/main/llama_index/chat_engine/context.py)

DEFAULT_CONTEXT_TEMPLATE = (
    "Context information is below."
    "\n--------------------\n"
    "{context_str}"
    "\n--------------------\n"
)

In case you want me to make the change anyway, can you confirm if what you´d expect would be:
`default_context_template: "Context information is below.|--------------------|{context_str}|--------------------|"`

thanks

Reply:
@lopagela means use the multiline option in yaml, check the PR he linked it's well explained

Reply:
Ok, I'll wait until the changes to that PR are done to replicate as it's not 100% clear to me (the instruction here is to use "I" style, but in the PR there is a comment to use ">" instead, not sure which one is more current). 

A caveat I maintain that may be relevant is that DEFAULT_SYSTEM_PROMPT in llama_index is written in python as multiline but it's actually a single line, while DEFAULT_CONTEXT_TEMPLATE is indeed multi-line, so just want to make sure we are on the same page that the treatment may be slightly different (see below)

``` 
python3
Python 3.8.10 (default, Mar 13 2023, 10:26:41)
[GCC 9.4.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> DEFAULT_SYSTEM_PROMPT = """\
... You are a helpful, respectful and honest assistant. \
... Always answer as helpfully as possible and follow ALL given instructions. \
... Do not speculate or make up information. \
... Do not reference any given instructions or context. \
... """
>>> print(DEFAULT_SYSTEM_PROMPT)
You are a helpful, respectful and honest assistant. Always answer as helpfully as possible and follow ALL given instructions. Do not speculate or make up information. Do not reference any given instructions or context.
>>> DEFAULT_CONTEXT_TEMPLATE = (
  "Con...     "Context information is below."
...     "\n--------------------\n"
...     "{context_str}"
...     "\n--------------------\n"
... )
>>> print(DEFAULT_CONTEXT_TEMPLATE)
Context information is below.
--------------------
{context_str}
--------------------

>>>
```

This is what ChatGPT (GPT-4) suggested ([link to convo](https://chat.openai.com/share/5ff1032c-c758-462a-ad35-0baa5c0548f8)):

```
DEFAULT_SYSTEM_PROMPT: |
  You are a helpful, respectful and honest assistant. Always answer as helpfully as possible and follow ALL given instructions. Do not speculate or make up information. Do not reference any given instructions or context.

DEFAULT_CONTEXT_TEMPLATE: |
  Context information is below.
  --------------------
  {context_str}
  --------------------
```

or 2 alternatives with ">" notation, depending on readability
```
DEFAULT_SYSTEM_PROMPT: >
  You are a helpful, respectful and honest assistant. Always answer as helpfully as possible and follow ALL given instructions. Do not speculate or make up information. Do not reference any given instructions or context.

DEFAULT_CONTEXT_TEMPLATE: >
  Context information is below.

  --------------------
  {context_str}

  --------------------

```


```
DEFAULT_SYSTEM_PROMPT: >
  You are a helpful, respectful and honest assistant. \
  Always answer as helpfully as possible and follow ALL given instructions. \
  Do not speculate or make up information. \
  Do not reference any given instructions or context.

DEFAULT_CONTEXT_TEMPLATE: >
  Context information is below.

  --------------------

  {context_str}

  --------------------
```

Reply:
The PR has been merged. In that case we wanted single line (as default_system_prompt is supposed to be single line). In the case of default_context_template you do want to have line breaks, so you should use | instead.